components/net/http_loader.rs

```diff
@@ -1603,17 +1634,20 @@ fn http_network_or_cache_fetch(
         // Step 2
         // TODO: Spec says requires testing on Proxy-Authenticate headers
 
-        // Step 3
-        // FIXME: Prompt the user for proxy authentication credentials
-
-        // Wrong, but will have to do until we are able to prompt the user
-        // otherwise this creates an infinite loop
-        // We basically pretend that the user declined to enter credentials
-        return response;
+        let loc_str = prompt_user_for_credentials(context).unwrap();
+        let temp_str = format!("Basic {}", loc_str);
+        http_request
+            .headers
+            .insert(header::AUTHORIZATION, temp_str.clone().parse().unwrap());
 
         // Step 4
-        // return http_network_or_cache_fetch(request, authentication_fetch_flag,
-        //                                    cors_flag, done_chan, context);
+        return http_network_or_cache_fetch(
+            http_request,
+            authentication_fetch_flag,
+            cors_flag,
+            done_chan,
+            context,
+        );
```

Comment:
These changes aren't quite right - [this step](https://fetch.spec.whatwg.org/#ref-for-proxy-authentication-entry%E2%91%A0) of the specification says to store the credentials somewhere and use them in [this step](https://fetch.spec.whatwg.org/#ref-for-proxy-authentication-entry), which is different than setting the Authorization header on the request. We should keep this code unmodified and add a FIXME comment saying that we can use the new prompt_user_for_credentials code but we need to store the result somewhere.